main/src/ca/uwaterloo/flix/language/phase/jvm/BackendObjType.scala

```diff
@@ -994,20 +994,27 @@ object BackendObjType {
       cm.mkField(ThreadsField)
       cm.mkConstructor(Constructor)
       cm.mkMethod(SpawnMethod)
+      cm.mkMethod(ExitMethod)
 
       cm.closeClassMaker()
     }
 
-    def ThreadsField: InstanceField = InstanceField(this.jvmName, IsPrivate, IsFinal, "threads", BackendObjType.ArrayList.toTpe)
+    // private ConcurrentLinkedQueue<Thread> threads = new ConcurrentLinkedQueue<Thread>();
+    def ThreadsField: InstanceField = InstanceField(this.jvmName, IsPrivate, IsFinal, "threads", BackendObjType.ConcurrentLinkedQueue.toTpe)
 
     def Constructor: ConstructorMethod = ConstructorMethod(this.jvmName, IsPublic, Nil, Some(
       thisLoad() ~ INVOKESPECIAL(JavaObject.Constructor) ~ 
-      thisLoad() ~ NEW(BackendObjType.ArrayList.jvmName) ~
-      DUP() ~ invokeConstructor(BackendObjType.ArrayList.jvmName, MethodDescriptor.NothingToVoid) ~
+      thisLoad() ~ NEW(BackendObjType.ConcurrentLinkedQueue.jvmName) ~
+      DUP() ~ invokeConstructor(BackendObjType.ConcurrentLinkedQueue.jvmName, MethodDescriptor.NothingToVoid) ~
       PUTFIELD(ThreadsField) ~
       RETURN()
     ))
 
+    // final public void spawn(Runnable r) {
```

Comment:
Any reason not to make the entire class final? (its not very important)

Reply:
Just that it didn't occur to me (blush)

Reply:
:)